### PR TITLE
chore: add parser to ESLint and fix offending Cypress code

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

This PR sets the stage for allowing us to use lint rules that require a TS compiler.

1. Adds a `parser` to our ESLint configuration
2. Includes the Cypress directory in TS compilation step so we catch build TS issues at both development and runtime
3. Incidentally, fixes current TS issues that snuck under the radar before type checking was set up for the Cypress directory
4. Removes a bunch of unused Cypress code

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A - code quality

## Risk

Minimal - no production changes.

## Testing

If CI still passes then Cypress runs correctly, and so does linting.

### Engineering

See the comment above.

### Operations

N/A

## Screenshots (if applicable)

N/A

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -10,6 +10,11 @@
     "prettier",
     "simple-import-sort"
   ],
+  "parser": "@typescript-eslint/parser",
```
Comment (gomesalexandre) [inline .eslintrc:13]: Interesting, this one is already present as a dependency of `@typescript-eslint/eslint-plugin` and docs recommend to explicitly add it as a parser. Looks like something we actually needed for some time already!

```shell
└─┬ react-scripts@5.0.0
  └─┬ eslint-config-react-app@7.0.0
    ├─┬ @typescript-eslint/eslint-plugin@5.14.0
    │ └── @typescript-eslint/parser@5.14.0 deduped
    └── @typescript-eslint/parser@5.14.0
```

File: tsconfig.json [hunk 2]
```diff
@@ -19,5 +19,5 @@
     "noUnusedParameters": true,
     "noUnusedLocals": true
   },
-  "include": ["src"]
+  "include": ["src", "cypress"]
```
Comment (gomesalexandre) [inline tsconfig.json:22]: Given we used `"noEmit": true`, confirmed this has no implications except for the sole purpose of type-checking

**Review by gomesalexandre — APPROVED** [review]
CI is happy, and local `yarn lint` ran as a sanity check is still working, so I'm happy

😄

File: .eslintrc (hunk 1 above)
Comment (0xApotheosis) [reply .eslintrc:13]: For sure, I think we are strictly better off with it.

**Pull request merged**